kobo/apps/accounts/forms.py

```diff
@@ -87,13 +89,20 @@ def __init__(self, *args, **kwargs):
         for field_name in list(self.fields.keys()):
             if field_name not in CONFIGURABLE_METADATA_FIELDS:
                 continue
-            if field_name not in desired_metadata_fields:
+            try:
+                desired_field = desired_metadata_fields[field_name]
+                if 'label' in desired_field.keys():
+                    try:
+                        self.fields[field_name].label = desired_field['label'][get_language()]
+                    except KeyError:
+                        self.fields[field_name].label = desired_field['label']['default']
```

Comment:
Ah, additional comment: it'd be good to have unit tests for these scenarios :)

Reply:
@jnm With the way I've been able to configure it [here](https://github.com/kobotoolbox/kpi/blob/bff0901589c91ef8cb511a8ea248158e6dddea44/kpi/fields/jsonschema_form_field.py#L88-L96), the label is an optional field, but Constance will not let you have the label without having the default field

Reply:
Good point about `default` :+1: There are some problems with the logic here, though: for example, because of the `continue` in
```python
if 'label' in desired_field.keys():
    …
else:
    continue
```
it's no longer possible to make a field required without also setting its label. I'll write a failing test for this and because you asked about testing methodology for these settings.

Reply:
Added to this branch: https://github.com/kobotoolbox/kpi/pull/4527/commits/3f96fa63abc757e7c854476cc42ef4745533b34a#diff-c6c76e400fd8f17237b0bafaff214dcf79f026bf9dcdd0c8eb34b4e09a9e6eabR40-R48